.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,78 @@
+name: deploy and integration test
+
+permissions:
+  id-token: write  # required for requesting the JWT
+  contents: read  # required for actions/checkout
+
+on:
+  workflow_dispatch:
+
+jobs:
+  deploy-and-integration-test:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Checkout repository with template stack
+      uses: actions/checkout@v3
+      with:
+        repository: developmentseed/eoapi-template
+        path: eoapi-template
```

Comment:
@emileten @sharkinsspatial I feel that having an external repo to host only the `tests` is maybe over complex. 

IMO we should have the integration tests code within the `.github/` directory OR have everything within the eoapi-tests repo (deploy + test) 

We already have a lot of `eoapi-*` repo and this might become confusing

Reply:
if we're going to re-use this we can add it directly on the root so other project can reuse it.

Reply:
ideally it would be cool if we could have a testing module as part of the `eoapi-cdk` library. But this will mean, rewrite everything in typescript instead of python.

maybe @alukach @sharkinsspatial have opinions 😅

Reply:
> if we're going to re-use this we can add it directly on the root so other project can reuse it.

root of `eoapi-cdk` works for now ? (the tests are _slightly_ coupled with AWS because of auth)

Reply:
Done, I moved the tests here

Reply:
@emileten Q: why do we need `eoapi-template`? IMO we should have everything we need here and not rely on a third party repo.

Reply:
🤔

Here, we define the constructs. If we want to integration-test, we need to instantiate and wrap all these constructs together. That's exactly what `eoapi-template` is.

Of course I don't think we want to copy-paste that wrapper code here.... 

If we don't like having too many moving repos, a solution would be to remove `eoapi-template` altogether and have its content simply in e.g. `eoapi-cdk/example`. 

@vincentsarago